docs/markdown/snippets/dep_as_static.md

```diff
@@ -0,0 +1,13 @@
+## `dep.as_static()` and `dep.as_shared()` functions
+
+Internal dependencies can now choose between static and shared version
+of a `both_libraries` linked library. This allows to use the same
+dependency objects for building either a static or a shared library from
+the same hierarchy of dependecies.
+
+`internal_depencency` now have `.as_static()` and `.as_shared()` methods, to
```

Comment:
You mean declare_dependency(), right?

Reply:
The methods belong to the object returned by `declare_dependency`. Not sure how I should identify it from the documentation point of view...